src/transformers/models/splinter/modeling_splinter.py

```diff
@@ -1029,18 +1029,14 @@ def forward(
         """
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
-        if question_positions is None:
-            if start_positions is not None and end_positions is not None:
-                raise TypeError(
-                    "For span selection training, question_positions must be specified to obtain the correct loss"
-                )
+        if question_positions is None and start_positions is not None and end_positions is not None:
+            raise TypeError("question_positions must be specified in order to calculate the loss")
 
-            if input_ids is not None:
-                question_positions = torch.argmax(
-                    (torch.eq(input_ids, self.question_token_id)).int(), dim=-1
-                ).unsqueeze(0)
-            else:
-                raise TypeError("question_positions is required when input_embeds is specified")
+        if question_positions is None and input_ids is None:
+            raise TypeError("question_positions must be specified when input_embeds is used")
+
+        if question_positions is None:
```

Comment:
```suggestion
        elif question_positions is None:
```

Reply:
Think it's easier to follow the logic like this, but feel free to disregard if you don't agree